### PR TITLE
Refactoring/aws/emr

### PR DESCRIPTION
**Pull request by Aboisier**
This PR migrated the EMR service to the new architecture, see #183 . There was a hack, caused by an API limitation. I kept it, since refactoring it is a bit out of the scope of this story.

### CODE REVIEW

File: ScoutSuite/providers/aws/facade/facade.py [hunk 1]
```diff
@@ -27,6 +28,8 @@ def __init__(self, credentials: dict = None):
         self.directconnect = DirectConnectFacade(self.session)
         self.efs = EFSFacade(self.session)
         self.elasticache = ElastiCacheFacade(self.session)
+        self.emr = EMRFacade(self.session)
+        self.emr = EMRFacade(self.session)
```
Comment (vifor2) [inline ScoutSuite/providers/aws/facade/facade.py]: Duplicate assignation

Comment (Aboisier) [reply ScoutSuite/providers/aws/facade/facade.py]: :o

Comment (Aboisier) [reply ScoutSuite/providers/aws/facade/facade.py]:
```suggestion
```

File: ScoutSuite/providers/aws/facade/emr.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from ScoutSuite.providers.aws.facade.basefacade import AWSBaseFacade
+from ScoutSuite.providers.utils import run_concurrently
+
+
+class EMRFacade(AWSBaseFacade):
+    async def get_clusters(self, region):
+        clusters_list = await AWSFacadeUtils.get_all_pages('emr', region, self.session, 'list_clusters', 'Clusters')
+        client = AWSFacadeUtils.get_client('emr', region, self.session)
+        clusters_descriptions = []
+        for cluster_id in [cluster['Id'] for cluster in clusters_list]:
```
Comment (misg) [inline ScoutSuite/providers/aws/facade/emr.py]: There's an API call behind each `client.describe_foobar`, so all `client.describe_foobar` calls should be done within `run_concurrently`. Thus, this loop should be done concurrently and parallelized like the following:

```suggestion
        cluster_ids = [cluster['Id'] for cluster in clusters_list]
        tasks = {
                asyncio.ensure_future(
                        run_concurrently(lambda: client.describe_cluster(ClusterId=cluster_id)['Cluster'])
                 ) for cluster_id in cluster_ids
        }
        for task in asyncio.as_completed(tasks):
                cluster = await task
                clusters_descriptions.append(cluster)
```

File: ScoutSuite/providers/aws/facade/emr.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+            cluster = client.describe_cluster(ClusterId=cluster_id)['Cluster']
```
Comment (misg) [inline ScoutSuite/providers/aws/facade/emr.py]:
```suggestion
```

File: ScoutSuite/providers/aws/facade/emr.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+            clusters_descriptions.append(cluster)
```
Comment (misg) [inline ScoutSuite/providers/aws/facade/emr.py]:
```suggestion
```

File: ScoutSuite/providers/aws/facade/emr.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+from ScoutSuite.providers.aws.facade.utils import AWSFacadeUtils
+from ScoutSuite.providers.aws.facade.basefacade import AWSBaseFacade
+from ScoutSuite.providers.utils import run_concurrently
+
```
Comment (misg) [inline ScoutSuite/providers/aws/facade/emr.py]:
```suggestion
import asyncio

```

**Pull request merged**